R/get_plot_component.R

```diff
@@ -0,0 +1,42 @@
+#' Get plot components
+#'
+#' @param plot a ggplot or gtable
+#' @param pattern the name of the component
+#'
+#' @return a grob
+#' @export
+get_plot_component <- function(plot, pattern) {
+
+  grob_names <- plot_component_names(plot)
+  grobs <- plot_components(plot)
```

Comment:
Here and throughout: The way the code is currently written, the plot is converted into a grob multiple times. Because the conversion is an expensive operation, it should be performed only once.